sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_polling.py

```diff
@@ -4,60 +4,184 @@
 # Licensed under the MIT License.
 # ------------------------------------
 
+from typing import Any, Union, Tuple, TYPE_CHECKING
+from azure.core.polling import LROPoller
 from azure.core.polling.base_polling import (
-    LongRunningOperation,
+    LROBasePolling,
+    OperationResourcePolling,
     _is_empty,
     _as_json,
     BadResponse,
-    OperationFailed
+    OperationFailed,
+    _raise_if_bad_http_status_and_method
 )
 
 from azure.core.exceptions import HttpResponseError, ODataV4Format
+from ._generated.models import TranslationStatus
+from ._models import JobStatusResult
 
 
-class TranslationPolling(LongRunningOperation):
-    """Implements a Location polling.
+if TYPE_CHECKING:
+    from azure.core.pipeline import PipelineResponse
+    from azure.core.pipeline.transport import (
+        HttpResponse,
+        AsyncHttpResponse,
+        HttpRequest,
+    )
+
+    ResponseType = Union[HttpResponse, AsyncHttpResponse]
+    PipelineResponseType = PipelineResponse[HttpRequest, ResponseType]
+
+
+_FINISHED = frozenset(["succeeded", "cancelled", "cancelling", "failed"])
+_FAILED = frozenset(["validationfailed"])
+
+
+class DocumentTranslationPoller(LROPoller):
+    """A custom poller implementation for Document Translation.
     """
 
-    def __init__(self):
-        self._async_url = None
+    @property
+    def id(self):
+        # type: () -> str
+        """The ID for the translation job
```

Comment:
I know you said renames later, and not sure if this is part of that, so wanted to call it out.
References of job instead of operation in these comments